redash/query_runner/activedata.py

```diff
@@ -0,0 +1,180 @@
+from redash.utils import JSONEncoder
+
+#originally from: https://github.com/klahnakoski/ActiveData-redash-query-runner/blob/c0e7286c09c6f1eb6746a6c7cca581bea79f4757/active_data.py
+
+logger = logging.getLogger(__name__)
+
+if not unicode:
+    unicode = str
+
+types_map = {
+    bool: TYPE_INTEGER,
+    str: TYPE_STRING,
+    unicode: TYPE_STRING,
+    dict: TYPE_STRING,
+    list: TYPE_STRING,
+    int: TYPE_INTEGER,
+    long: TYPE_INTEGER,
+    float: TYPE_FLOAT
+}
+
+
+class ActiveData(BaseSQLQueryRunner):
+    noop_query = "SELECT 1"
+
+    def __init__(self, configuration):
+        super(ActiveData, self).__init__(configuration)
+
+    @classmethod
+    def configuration_schema(cls):
+        return {
+            "type": "object",
+            "properties": {
+                "host_url": {
+                    "type": "string",
+                    "title": "Host URL",
+                    "default": "https://activedata.allizom.org:80",
+                    "info": "Do not end with a trailing slash."
+                },
+                "doc_url": {
+                    "type": "string",
+                    "title": "Documentation URL",
+                    "default": "https://github.com/klahnakoski/ActiveData/tree/dev/docs"
+                }
+            },
+            "required": ["host_url"]
+        }
+
+    @classmethod
+    def name(cls):
+        return "ActiveData"
+
+    @classmethod
+    def type(cls):
+        return "activedata"
+
+    @classmethod
+    def enabled(cls):
+        return True
+
+    def _get_tables(self, schema):
+        query = {
+            "from": "meta.columns",
+            "select": [
+                "name",
+                "table"
+            ],
+            "where": {"not": {"prefix": {"es_index": "meta."}}},
+            "limit": 1000,
+            "format": "list"
+        }
+        results = self.run_jx_query(query, None)
+
+        for row in results['data']:
+            table_name = row['table']
+
+            if table_name not in schema:
+                schema[table_name] = {'name': table_name, 'columns': []}
+
+            schema[table_name]['columns'].append(row['name'])
+
+        return [{'name': r['name'], 'columns': sorted(r['columns'])} for r in schema.values()]
+
+    def run_jx_query(self, query, user):
+        data = json.dumps(query, ensure_ascii=False)
+        result = requests.post(self.configuration['host_url']+"/query", data=data)
+        response = json.loads(result.content)
+
+        if response.get('type') == "ERROR":
+            cause = find_cause(response)
+            raise Exception(cause)
+        return response
+
+    def run_query(self, annotated_query, user):
+        request = {}
+        comment, request["sql"] = annotated_query.split("*/", 2)
+        meta = request['meta'] ={}
+        for kv in comment.strip()[2:].split(","):
+            k, v = map(unicode.strip, kv.split(":"))
```

Comment:
`unicode` is used here; it must be changed, but how it changes depends on the version of Python.